File: WorldFeed/Client/src/app/core/Core/DTO/ExtendedTweet.ts
import {ITweetEntities} from "../../Public/Models/Entities/ITweetEntities";
import {IExtendedTweet} from "../../Public/Models/Interfaces/DTO/IExtendedTweet";

export class ExtendedTweet implements IExtendedTweet {
  // [JsonProperty("text")]
  public text: string;

  // [JsonProperty("full_text")]
  public fullText: string;

  // [JsonProperty("display_text_range")]
  public displayTextRange: number[];

  // [JsonProperty("entities")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public legacyEntities: ITweetEntities;

  // [JsonProperty("extended_entities")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public extendedEntities: ITweetEntities;
}

File: Sharebook/Server/src/Services/Post/Sharebook.Post.API/DTO/ExtendedTweet.cs
namespace Sharebook.Post.DTO
{
    using Newtonsoft.Json;

    using Sharebook.Common.JsonConverters;
    using Sharebook.Common.Public.Models.Entities;
    using Sharebook.Common.Public.Models.Interfaces.DTO;

    public class ExtendedTweet : IExtendedTweet
    {
        [JsonProperty("text")]
        public string Text { get; set; }

        [JsonProperty("full_text")]
        public string FullText { get; set; }

        [JsonProperty("display_text_range")]
        public int[] DisplayTextRange { get; set; }

        [JsonProperty("entities")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public ITweetEntities LegacyEntities { get; set; }

        [JsonProperty("extended_entities")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public ITweetEntities ExtendedEntities { get; set; }
    }
}
